tests/base.py

```diff
@@ -97,71 +68,90 @@ def teardown_method(self, test_method):
+        "Wrap a raw event like the Sentry codebase does before sending to Kafka."
+
+        unique = "%s:%s" % (str(data["project"]), data["id"])
+        primary_hash = md5(unique.encode("utf-8")).hexdigest()
+
+        return InsertEvent(
+            {
+                "event_id": data["id"],
+                "group_id": int(primary_hash[:16], 16),
+                "primary_hash": primary_hash,
+                "project_id": data["project"],
+                "message": data["message"],
+                "platform": data["platform"],
+                "datetime": data["datetime"],
+                "data": data,
+                "organization_id": data["organization_id"],
+                "retention_days": settings.DEFAULT_RETENTION_DAYS,
+            }
+        )
 
-    def create_event_for_date(self, dt, retention_days=settings.DEFAULT_RETENTION_DAYS):
-        event = {
+    def __get_event(self) -> InsertEvent:
+        from tests.fixtures import raw_event
+
+        timestamp = datetime.utcnow()
+        return self.__wrap_event_data(
+            {
+                "datetime": (timestamp - timedelta(seconds=2)).strftime(
+                    settings.PAYLOAD_DATETIME_FORMAT,
+                ),
+                "received": int(
+                    calendar.timegm((timestamp - timedelta(seconds=1)).timetuple())
+                ),
+                **deepcopy(raw_event),
+            }
+        )
+
+    def create_event_row_for_date(
+        self, dt: datetime, retention_days=settings.DEFAULT_RETENTION_DAYS
+    ):
+        return {
             "event_id": uuid.uuid4().hex,
             "project_id": 1,
             "group_id": 1,
             "deleted": 0,
+            "timestamp": dt,
+            "retention_days": retention_days,
         }
-        event["timestamp"] = dt
-        event["retention_days"] = retention_days
-        return event
-
-    def write_raw_events(self, events):
-        if not isinstance(events, (list, tuple)):
-            events = [events]
-
-        out = []
-        for event in events:
-            if "primary_hash" not in event:
-                event = wrap_raw_event(event)
-            processed = (
-                enforce_table_writer(self.dataset)
-                .get_stream_loader()
-                .get_processor()
-                .process_message(event)
-            )
-            out.extend(processed.data)
-        return self.write_processed_records(out)
 
-    def write_processed_events(self, events):
-        if not isinstance(events, (list, tuple)):
-            events = [events]
+    def write_events(self, events: Sequence[Union[EventData, InsertEvent]]) -> None:
+        processor = (
+            enforce_table_writer(self.dataset).get_stream_loader().get_processor()
+        )
 
-        rows = []
+        processed_messages = []
         for event in events:
-            rows.append(event)
+            if self.__is_event_data(event):
```

Comment:
Is there actually a legit case where this may return false and the test should not fail ?
Can't we assert this?

Reply:
I thought there still was, but I guess I actually changed all of the inputs to this function to `InsertEvent` and didn't even realize it. Cool, I'll remove this conditional. (I also felt like `__is_event_data` was a super hacky function so it's nice to be able to remove it as well.)

Reply:
Cleaned up here: b23b209a608619f5d7481deac60c77d270deb7af

It's not very easy to assert on whether or something satisfies a `TypedDict` at runtime so going to punt on that for now, I'll likely be changing this into a different data structure soon enough as well (foreshadowing.…)